src/ConnectedMode/Suppressions/IssueServerEventsListener.cs

```diff
@@ -0,0 +1,91 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration;
+using SonarQube.Client.Models.ServerSentEvents.ClientContract;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Suppressions
+{
+    /// <summary>
+    /// Handles <see cref="IIssueServerEvent"/> coming from the server.
+    /// </summary>
+    internal interface IIssueServerEventsListener : IDisposable
+    {
+        Task ListenAsync();
+    }
+
+    [Export(typeof(IIssueServerEventsListener))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class IssueServerEventsListener : IIssueServerEventsListener
+    {
+        private readonly IIssueServerEventSource issueServerEventSource;
+        private readonly IThreadHandling threadHandling;
+        private readonly ILogger logger;
+        private readonly CancellationTokenSource cancellationTokenSource;
+
+        [ImportingConstructor]
+        public IssueServerEventsListener(IIssueServerEventSource issueServerEventSource,
+            IThreadHandling threadHandling,
+            ILogger logger)
+        {
+            this.issueServerEventSource = issueServerEventSource;
+            this.threadHandling = threadHandling;
+            this.logger = logger;
+
+            cancellationTokenSource = new CancellationTokenSource();
+        }
+
+        public async Task ListenAsync()
+        {
+            await threadHandling.SwitchToBackgroundThread();
+
+            while (!cancellationTokenSource.IsCancellationRequested)
+            {
+                try
+                {
+                    var issueServerEvent = await issueServerEventSource.GetNextEventOrNullAsync();
+
+                    if(issueServerEvent == null)
+                    {
+                        // Will return null when issueServerEventSource is disposed
+                        return;
+                    }
+
+                    // TODO: Implement handling of event.
+                }
+                catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+                {
+                    logger.LogVerbose($"[IssueServerEventsListener] Failed to handle issue event: {ex}");
+                }
+            }
+        }
+
+        public void Dispose()
+        {
+            cancellationTokenSource.Cancel();
+            cancellationTokenSource?.Dispose();
+        }
+    }
```

Comment:
There's a lot of overlap between the taint and issue tests and implementation.
We should consider refactoring later to create a base class if it turns out that the only differences in behaviour are effectively in line 76 when recognised events are processed.